### PR TITLE
Avoid using panic for user-facing errors. 

### LINKED ISSUE #1799: Avoid using `panic` for user-facing errors.
**Issue summary**

Panics are for internal errors and should never intentionally be shown to the end-user. The following six panics should be replaced with proper error handling:

- https://github.com/ChainSafe/forest/blob/7e470572033e269d75efbebae1579dcd2bebf35b/forest/src/cli/state_cmd.rs#L54
- https://github.com/ChainSafe/forest/blob/7e470572033e269d75efbebae1579dcd2bebf35b/forest/src/cli/state_cmd.rs#L104
- https://github.com/ChainSafe/forest/blob/7e470572033e269d75efbebae1579dcd2bebf35b/forest/src/cli/state_cmd.rs#L148
- https://github.com/ChainSafe/forest/blob/7e470572033e269d75efbebae1579dcd2bebf35b/forest/src/cli/fetch_params_cmd.rs#L33
- https://github.com/ChainSafe/forest/blob/7e470572033e269d75efbebae1579dcd2bebf35b/forest/src/cli/mod.rs#L209
- https://github.com/ChainSafe/forest/blob/7e470572033e269d75efbebae1579dcd2bebf35b/forest/src/cli/mod.rs#L344

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1799

**Other information and links**

### CODE REVIEW

File: forest/src/cli/mod.rs
Comment (lemmih) [inline forest/src/cli/mod.rs:355]: You could use `map_err` and `?`.

File: forest/src/cli/mod.rs
Comment (lemmih) [inline forest/src/cli/mod.rs:466]: Maybe something like this:

```suggestion
            assert_eq!(to_size_string(&input), Ok(expected.to_string()));
```

File: forest/src/cli/mod.rs
Comment (lemmih) [inline forest/src/cli/mod.rs:473]: This might be more idiomatic:

```suggestion
        assert!(to_size_string(&BigInt::from(-1i8)).is_err());
```

**Review by lemmih — APPROVED** [review]
Nice. I do wonder if the `run` method should return a `Result`. Hmm, we can think about that in another PR.

File: forest/src/cli/mod.rs
Comment (LesnyRumcajs) [inline forest/src/cli/mod.rs:345]: Perhaps let's use `anyhow::Result`?

File: forest/src/cli/mod.rs
Comment (LesnyRumcajs) [inline forest/src/cli/mod.rs:347]: do we need `return` here?

Comment (LesnyRumcajs) [inline forest/src/cli/mod.rs:347]: why do we need to clone those values, especially the error?

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]

File: forest/src/cli/mod.rs
Comment (tyshko5) [reply forest/src/cli/mod.rs]: Deleted

Comment (tyshko5) [reply forest/src/cli/mod.rs]: Deleted

File: forest/src/cli/mod.rs
Comment (tyshko5) [reply forest/src/cli/mod.rs:345]: For now we are using basic `Result` everywhere there, so I use it too

Comment (connormullett) [reply forest/src/cli/mod.rs:345]: `anyhow::Result` potentially could help us with those nasty map_err statements. We use regularly all over the codebase but this could be a separate issue. @lemmih thoughts?

**Review by connormullett — APPROVED** [review]
Aside from one comment, looks good. Approving to not hold up after next reviewer

**Pull request merged**
